### PR TITLE
Use cabal-docspec

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by phadej** [comment]
This turned to be a breaking change, but I think its better to have uniform API

File: src/Data/LCA/Online.hs [hunk 1]
```diff
@@ -56,12 +57,13 @@ import Data.LCA.View
...
 import Prelude hiding
   ( drop
-#if __GLASGOW_HASKELL__ < 710
   , length
   , null
-#endif
   )
 
+-- $setup
+-- >>> let length = Data.LCA.Online.length
```
Comment (RyanGlScott) [inline src/Data/LCA/Online.hs:65]: Why does this need to be here if `length` is hidden in the imports?

File: src/Data/LCA/Online/Naive.hs [hunk 2]
```diff
@@ -142,6 +140,10 @@ _                ~= _                = False
 {-# INLINE (~=) #-}
 
 -- | /O(h)/ Compute the lowest common ancestor of two paths
+--
+-- >>> let fromList' = fromList . map (flip (,) ())
```
Comment (RyanGlScott) [inline src/Data/LCA/Online/Naive.hs:144]: Personally, I find `\x -> (x, ())` easier to read than `flip (,) ()`, but I don't feel strongly about this.

**Review by RyanGlScott — COMMENTED** [review]
> This turned to be a breaking change, but I think its better to have uniform exported API

I'm not following. In what sense is this a breaking change?

File: src/Data/LCA/Online/Naive.hs (hunk 2 above)
Comment (phadej) [reply src/Data/LCA/Online/Naive.hs:144]: It's the same example as in other modules. I simply copied that. I could change all.

File: src/Data/LCA/Online.hs (hunk 1 above)
Comment (phadej) [reply src/Data/LCA/Online.hs:65]: `cabal-docspec` doesn't see imports. It `:m +Data.LCA.Online` and then the example uses `length` which then conflicts with `Prelude.length`.

**Comment by phadej** [comment]
> I'm not following. In what sense is this a breaking change?

Currently `lca` exports `null` and `length` specific to `Path` on GHC-7.8 and older, and `Data.Foldable.null/length` on GHC-7.10 and newer. This patch makes `lca` always export the specific ones.

File: src/Data/LCA/Online/Naive.hs (hunk 2 above)
Comment (RyanGlScott) [reply src/Data/LCA/Online/Naive.hs:144]: Ah, I overlooked the fact that this is defined elsewhere in the library. In that case, I agree that it's best to just copy it for the sake of this PR.

File: src/Data/LCA/Online.hs (hunk 1 above)
Comment (RyanGlScott) [reply src/Data/LCA/Online.hs:65]: Ah, OK—I wasn't aware of that nuance of `cabal-docspec`.

**Review by RyanGlScott — COMMENTED** [review]
> Currently `lca` exports `null` and `length` specific to `Path` on GHC-7.8 and older, and `Data.Foldable.null/length` on GHC-7.10 and newer. This patch makes `lca` always export the specific ones.

I see, thanks for clarifying. Can you include a mention of this in the changelog and bump the version number accordingly?

**Comment by phadej** [comment]
> > Currently `lca` exports `null` and `length` specific to `Path` on GHC-7.8 and older, and `Data.Foldable.null/length` on GHC-7.10 and newer. This patch makes `lca` always export the specific ones.
> 
> I see, thanks for clarifying. Can you include a mention of this in the changelog and bump the version number accordingly?

Done.

**Comment by phadej** [comment]
> Ah, OK—I wasn't aware of that nuance of cabal-docspec.

`cabal-docspec` is truly "doc". It doesn't even try to look into an implementation for some secret bits, only to find `>>> examples`. In this sense it is more principled.  (My vision that in year 2030 it would rather use `:doc` command to find examples to test).

**Pull request merged**